tests/functional/l/len_checks.py

```diff
@@ -102,3 +102,50 @@ def github_issue_1331_v4(*args):
 
 b = bool(len(z)) # [len-as-condition]
 c = bool(len('TEST') or 42) # [len-as-condition]
+
+
+def github_issue_1879():
+
+    class ClassWithBool(list):
+        def __bool__(self):
+            return True
+
+    class ClassWithoutBool(list):
+        pass
+
+    class ChildClassWithBool(ClassWithBool):
+        pass
+
+    class ChildClassWithoutBool(ClassWithoutBool):
+        pass
+
+    assert len(ClassWithBool())
+    # We could expect to not have a len-as-condition for ChildClassWithBool,
+    # but I don't think the required analysis is worth it.
```

Comment:
I think we could achieve it quite easily with something like:
```
    @staticmethod
    def instance_has_bool(class_def: astroid.ClassDef) -> bool:
        try:
            class_def.getattr('__bool__')
            return True
        except astroid.AttributeInferenceError:
            return False
```

Reply:
Haha yeah it's a lot better. It also fixes a test case I wasn't doing right ! https://github.com/PyCQA/pylint/pull/3821/commits/41ad660a1c230118b47c5cbb7685372ce9b3c5e6